src/server.jl

```diff
@@ -338,11 +344,12 @@ function ws_upgrade(http::HTTP.Stream)
     HTTP.setheader(http, "Upgrade" => "websocket")
     HTTP.setheader(http, "Connection" => "Upgrade")
     key = HTTP.header(http, "Sec-WebSocket-Key")
-    HTTP.setheader(http, "Sec-WebSocket-Accept" => HTTP.WebSockets.accept_hash(key))
+    HTTP.setheader(http, "Sec-WebSocket-Accept" => HTTP.WebSockets.hashedkey(key))
     HTTP.startwrite(http)
 
     io = http.stream
-    return HTTP.WebSockets.WebSocket(io; server=true)
+    return HTTP.WebSocket(io, http.message.request, http.message; client=false)
+    # return HTTP.WebSockets.WebSocket(io; server=true)
```

Comment:
@quinnj can you advice on this?

Reply:
Yes, this looks fine to me.

Reply:
I mean, it doesn't work :D

Reply:
In what way does it not work? What kind of error are we seeing?

Reply:
I can't fetch the request field like that IIRC (not by a computer atm).

Reply:
In the `WebSockets.upgrade` code, we're doing:
```julia
req = http.message
ws = WebSocket(io, req, req.response; client=false, maxframesize, maxfragmentation)
```
So perhaps we need to do
```suggestion
    return HTTP.WebSocket(io, http.message, http.message.response; client=false)
```